cmake/BundledOpenSSL.cmake

```diff
@@ -59,7 +65,8 @@ function(use_openssl SOURCE_DIR BINARY_DIR)
             no-module           # disable dynamically loadable engines
             no-pinshared        # don't pin shared libraries in the process memory
             enable-tfo          # Enable TCP Fast Open
-            no-ssl)             # disable SSLv3
+            no-ssl              # disable SSLv3
+            no-engine)
```

Comment:
What does this exactly do? INSTALL.md says:
> Don't build support for loading engines.

But I'm not sure what loadable engines are. Is it related to FIPS in any way? Also, the FIPS target omits most of these, probably leading to some unnecessary extra build steps, consider adding some more options there, like no-tests, no-docs.

Reply:
I'm not that familiar with it, but there is an Engine API in OpenSSL that is controlled by this option. An engine that is loaded in OpenSSL is a hardware or software implementation used for performing cryptographic operations. Engines are not FIPS compatible and corresponding API is deprecated since OpenSSL 3.0, so for FIPS compliancy we should remove the engine support as well. (According to the documention it is also a good guidline for FIPS compliancy to remove anything that is deprecated in OpenSSL 3.0 as that is probably not FIPS compatible.)